modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/TypedInput.es.js

```diff
@@ -97,24 +100,89 @@ class TypedInput extends React.Component {
 	_handleSelectEntity = () => {
 		const {
 			onChange,
-			selectEntity: {id, title, uri}
+			selectEntity: {id, multiple, title, uri},
+			value
 		} = this.props;
 
-		Liferay.Util.selectEntity(
-			{
-				dialog: {
-					constrain: true,
-					destroyOnHide: true,
-					modal: true
+		if (multiple) {
```

Comment:
Fantastic. Although, it'd be great to start extracting this big chucks of functionality into functions. Specially the ones related to out-of-react stuff. This simple component is getting to 300 lines. Just a suggestion.